sklearn/compose/_column_transformer.py

```diff
@@ -512,7 +512,8 @@ def _hstack(self, Xs):
         Xs : List of numpy arrays, sparse arrays, or DataFrames
         """
         if self.sparse_output_:
-            return sparse.hstack(Xs).tocsr()
+            return sparse.hstack([check_array(X, accept_sparse=True)
```

Comment:
this may be too strict in terms of dtype and finiteness checks...?

Reply:
I *think* it might be OK. Looking at the code of `check_array`, if `dtype="numeric"` (the default), it only seems to convert the array to float if it is object, and otherwise it leaves the dtype alone (which is I think what we want here).

But I agree that finiteness should not be checked, so adding a `force_all_finite=False` is probably what we want.